src/components/TodoItem.tsx

```diff
@@ -0,0 +1,101 @@
+import cn from 'classnames';
+import { FC, useCallback, useState } from 'react';
+import { Todo } from '../types/Todo';
+import { EditForm } from './EditForm';
+
+interface Props {
+  todo: Todo;
+  removeTodoByID?: (arg: number) => void;
+  editTodoByID?: (id: number, data: Partial<Todo>) => Promise<void>;
+  isPresentInLoadingTodos?: boolean;
+}
+
+export const TodoItem:FC<Props> = ({
+  todo,
+  removeTodoByID = () => {},
+  editTodoByID = () => {},
+  isPresentInLoadingTodos = false,
+}) => {
+  const { title, completed, id } = todo;
+  const [isLoading, setIsLoading] = useState(false);
+  const [isEditing, setIsEditing] = useState(false);
+
+  const onHandleRemoveTodo = useCallback(async () => {
+    setIsLoading(true);
+    await removeTodoByID(id);
+    setIsLoading(false);
+  }, [id, removeTodoByID]);
+
+  const editCurrentTodo = useCallback(async (data: Partial<Todo>) => {
+    setIsLoading(true);
+    await editTodoByID(id, data);
```

Comment:
Any asynchronous requests should be wrapped in a `try-catch` block to handle unexpected errors and prevent application crashes.